### ISSUE TITLE
xArray doesn't copy per-variable attributes when subsetting

### ISSUE THREAD

**Issue by htmlboss** [DFO-Ocean-Navigator/Ocean-Data-Map-Project#655]
See `votemper` below for example. This may cause some hang-ups down the line for users of the subset files.

Original:
```sh
netcdf \2020012100_240_3D_ps5km60N {
dimensions:
	xc = 1770 ;
	yc = 1610 ;
	depth = 50 ;
	time = 1 ;
variables:
	float xc(xc) ;
		xc:units = "m" ;
		xc:long_name = "x-coordinate of polar-stereographic projection" ;
		xc:standard_name = "projection_x_coordinate" ;
		xc:axis = "X" ;
	float yc(yc) ;
		yc:units = "m" ;
		yc:long_name = "y-coordinate of polar-stereographic projection" ;
		yc:standard_name = "projection_y_coordinate" ;
		yc:axis = "Y" ;
	float latitude(yc, xc) ;
		latitude:units = "degrees_north" ;
		latitude:valid_min = -90.f ;
		latitude:valid_max = 90.f ;
		latitude:long_name = "Latitude" ;
		latitude:standard_name = "latitude" ;
	float longitude(yc, xc) ;
		longitude:units = "degrees_east" ;
		longitude:valid_min = 0.f ;
		longitude:valid_max = 360.f ;
		longitude:long_name = "Longitude" ;
		longitude:standard_name = "longitude" ;
	char polar_stereographic ;
		polar_stereographic:grid_mapping_name = "polar_stereographic" ;
		polar_stereographic:earth_radius = 6371229.f ;
		polar_stereographic:latitude_of_projection_origin = 90.f ;
		polar_stereographic:straight_vertical_longitude_from_pole = -100.f ;
		polar_stereographic:scale_factor_at_projection_origin = 0.9330124f ;
		polar_stereographic:false_easting = 4245000.f ;
		polar_stereographic:false_northing = 5295000.f ;
	float depth(depth) ;
		depth:units = "m" ;
		depth:valid_min = 0.f ;
		depth:valid_max = 5875.f ;
		depth:long_name = "depth at T-grid points" ;
		depth:standard_name = "depth" ;
		depth:axis = "Z" ;
		depth:positive = "down" ;
	double time(time) ;
		time:calendar = "gregorian" ;
		time:units = "seconds since 1950-01-01 00:00:00" ;
		time:long_name = "Validity time" ;
		time:standard_name = "time" ;
		time:axis = "T" ;
	float votemper(time, depth, yc, xc) ;
		votemper:units = "Kelvin" ;
		votemper:_FillValue = 1.e+20f ;
		votemper:missing_value = 1.e+20f ;
		votemper:valid_min = 173.f ;
		votemper:valid_max = 373.f ;
		votemper:long_name = "Sea water potential temperature" ;
		votemper:short_name = "votemper" ;
		votemper:standard_name = "sea_water_potential_temperature" ;
		votemper:coordinates = "longitude latitude time" ;
		votemper:cell_methods = "time: mean (interval: 24 hours)" ;
		votemper:grid_mapping = "polar_stereographic" ;
		votemper:nomvar = "TM" ;
		votemper:stdscale = 1.f ;
		votemper:stdoffset = 0.f ;
	float vosaline(time, depth, yc, xc) ;
		vosaline:units = "1e-3" ;
		vosaline:_FillValue = 1.e+20f ;
		vosaline:missing_value = 1.e+20f ;
		vosaline:valid_min = 0.f ;
		vosaline:valid_max = 45.f ;
		vosaline:long_name = "Sea water salinity" ;
		vosaline:short_name = "vosaline" ;
		vosaline:standard_name = "sea_water_salinity" ;
		vosaline:coordinates = "longitude latitude time" ;
		vosaline:cell_methods = "time: mean (interval: 24 hours)" ;
		vosaline:grid_mapping = "polar_stereographic" ;
		vosaline:nomvar = "SALW" ;
		vosaline:stdscale = 1.f ;
		vosaline:stdoffset = 0.f ;
	float vozocrtx(time, depth, yc, xc) ;
		vozocrtx:units = "m s-1" ;
		vozocrtx:_FillValue = 1.e+20f ;
		vozocrtx:missing_value = 1.e+20f ;
		vozocrtx:valid_min = -20.f ;
		vozocrtx:valid_max = 20.f ;
		vozocrtx:long_name = "Sea water x velocity" ;
		vozocrtx:short_name = "vozocrtx" ;
		vozocrtx:standard_name = "sea_water_x_velocity" ;
		vozocrtx:coordinates = "longitude latitude time" ;
		vozocrtx:cell_methods = "time: mean (interval: 24 hours)" ;
		vozocrtx:grid_mapping = "polar_stereographic" ;
		vozocrtx:nomvar = "UUW" ;
		vozocrtx:stdscale = 1.f ;
		vozocrtx:stdoffset = 0.f ;
	float vomecrty(time, depth, yc, xc) ;
		vomecrty:units = "m s-1" ;
		vomecrty:_FillValue = 1.e+20f ;
		vomecrty:missing_value = 1.e+20f ;
		vomecrty:valid_min = -20.f ;
		vomecrty:valid_max = 20.f ;
		vomecrty:long_name = "Sea water y velocity" ;
		vomecrty:short_name = "vomecrty" ;
		vomecrty:standard_name = "sea_water_y_velocity" ;
		vomecrty:coordinates = "longitude latitude time" ;
		vomecrty:cell_methods = "time: mean (interval: 24 hours)" ;
		vomecrty:grid_mapping = "polar_stereographic" ;
		vomecrty:nomvar = "VVW" ;
		vomecrty:stdscale = 1.f ;
		vomecrty:stdoffset = 0.f ;
// global attributes:
		:Conventions = "CF-1.6" ;
		:title = "Time-averaged sea ice and ocean forecast fields" ;
		:institution = "The Canadian Centre for Meteorological and Environmental Prediction" ;
		:source = "Global Ice Ocean Prediction System version 3.0.0" ;
		:product_version = "GIOPS_3.0.0_F3" ;
		:creation_date = "2020-Jan-21 04:50:19 UTC" ;
		:contact = "production-info@ec.gc.ca" ;
}
```
Subset:
```sh
netcdf giops_test {
dimensions:
	depth = 50 ;
	time = 1 ;
	xc = 61 ;
	yc = 203 ;
variables:
	double depth(depth) ;
		depth:_FillValue = NaN ;
	double time(time) ;
		time:_FillValue = NaN ;
	double xc(xc) ;
		xc:_FillValue = NaN ;
	double yc(yc) ;
		yc:_FillValue = NaN ;
	float latitude(yc, xc) ;
		latitude:_FillValue = NaNf ;
		latitude:units = "degrees_north" ;
		latitude:valid_min = -90.f ;
		latitude:valid_max = 90.f ;
		latitude:long_name = "Latitude" ;
		latitude:standard_name = "latitude" ;
	float longitude(yc, xc) ;
		longitude:_FillValue = NaNf ;
		longitude:units = "degrees_east" ;
		longitude:valid_min = 0.f ;
		longitude:valid_max = 360.f ;
		longitude:long_name = "Longitude" ;
		longitude:standard_name = "longitude" ;
	float votemper(time, depth, yc, xc) ;
		votemper:_FillValue = NaNf ;
		votemper:coordinates = "longitude latitude" ;
// global attributes:
		:Conventions = "CF-1.6" ;
		:title = "Time-averaged sea ice and ocean forecast fields" ;
		:institution = "The Canadian Centre for Meteorological and Environmental Prediction" ;
		:source = "Global Ice Ocean Prediction System version 7.1.0" ;
		:product_version = "GIOPS_7.1.0_F3" ;
		:creation_date = "2020-Feb-12 04:50:24 UTC" ;
		:contact = "production-info@ec.gc.ca" ;
}
```

Docs: http://xarray.pydata.org/en/stable/indexing.html

Is this the intended xArray behavior (i.e. do we need to explicitly copy these attributes afterwards)?

**Comment by htmlboss** [comment]
The issue has to do with calculated variables not being assigned any attributes on creation of the DataArray

**Comment by JustinElms** [comment]
Fixed with PRs #924 and #925.
